pkg/deploy/config.go

```diff
@@ -129,6 +129,11 @@ type Provider struct {
 	With map[string]string `yaml:"with" json:"with"`
 }
 
+type NotificationsMap struct {
```

Comment:
I would like us to rename this to Notifications rather than NotificationsMap because it is not a map type and does not behave like one.
Could you change this everywhere, you can use rename symbol in vscode
```suggestion
type Notifications struct {
```